lib/response.js

```diff
@@ -82,8 +82,8 @@ module.exports = {
   set status(code) {
     if (this.headerSent) return;
 
-    assert('number' == typeof code, 'status code must be a number');
-    assert(statuses[code], `invalid status code: ${code}`);
+    if ('number' !== typeof code) throw new TypeError(util.format('non-numeric status code: %j', code));
```

Comment:
why util.format instead of string interpolation?

Reply:
There is no official reason to reject string interpolation. The results of the followed forms are same.
```javascript
new TypeError(util.format('non-numeric status code: %j', code));
new TypeError(`non-numeric status code: ${JSON.stringify(code)}`);
```
But I'd like to consider `util.format` as a pattern cuz it has been used in many places: 
https://github.com/koajs/koa/blob/77a4cfb8297bed0bc577425a6f42a0397547ca32/lib/context.js#L109
https://github.com/koajs/koa/blob/77a4cfb8297bed0bc577425a6f42a0397547ca32/lib/application.js#L188

Reply:
Cool, I can make another pull request to change that in a different PR some day in the future.